src/fevm.rs

```diff
@@ -0,0 +1,130 @@
+use anyhow::anyhow;
+use fvm::executor::{ApplyKind, Executor};
+use fvm_integration_tests::dummy::DummyExterns;
+use fvm_integration_tests::tester::{Account, Tester};
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::{strict_bytes, to_vec, tuple::*, BytesDe, BytesSer, Cbor, RawBytes};
+use fvm_shared::{address::Address, message::Message, ActorID, METHOD_CONSTRUCTOR};
+
+pub fn run<B: Blockstore>(
+    tester: &mut Tester<B, DummyExterns>,
+    contract: &[u8],
+    entrypoint: &[u8],
+    params: &[u8],
+) -> anyhow::Result<()> {
+    let accounts: [Account; 1] = tester.create_accounts().unwrap();
+    tester
+        .instantiate_machine_with_config(DummyExterns, |cfg| cfg.actor_debugging = false)
+        .unwrap();
+
+    // create actor
+    let create_params = Create2Params {
+        initcode: Vec::from(contract),
+        salt: [0u8; 32],
+    };
+    let create_params_ser = to_vec(&create_params).unwrap();
+    let create_mlen = create_params_ser.len();
+    let create_msg = Message {
+        from: accounts[0].1,
+        to: Address::new_id(10),
+        gas_limit: 10_000_000_000,
+        method_num: EAMMethod::Create2 as u64,
+        params: RawBytes::from(create_params_ser),
+        ..Message::default()
+    };
+
+    let create_res = tester
+        .executor
+        .as_mut()
+        .unwrap()
+        .execute_message(create_msg, ApplyKind::Explicit, create_mlen)
+        .unwrap();
+
+    if create_res.msg_receipt.exit_code.value() != 0 {
+        return Err(anyhow!(
+            "actor creation failed: {}",
+            create_res.msg_receipt.exit_code
+        ));
+    }
+
+    let create_return: CreateReturn = create_res.msg_receipt.return_data.deserialize().unwrap();
+
+    // invoke contract
+    let mut input_data = Vec::from(entrypoint);
+    let mut input_params = Vec::from(params);
+    input_data.append(&mut input_params);
+    let invoke_msg = Message {
+        from: accounts[0].1,
+        to: Address::new_id(create_return.actor_id),
+        sequence: 1,
+        gas_limit: 10_000_000_000,
+        method_num: EVMMethod::InvokeContract as u64,
+        params: RawBytes::serialize(BytesSer(&input_data)).unwrap(),
+        ..Message::default()
+    };
+    let invoke_mlen = invoke_msg.params.len();
+
+    let invoke_res = tester
+        .executor
+        .as_mut()
+        .unwrap()
+        .execute_message(invoke_msg, ApplyKind::Explicit, invoke_mlen)
+        .unwrap();
+
+    if invoke_res.msg_receipt.exit_code.value() != 0 {
+        return Err(anyhow!(
+            "contract invocation failed: {}",
+            create_res.msg_receipt.exit_code
+        ));
+    }
+
+    let BytesDe(invoke_result) = invoke_res.msg_receipt.return_data.deserialize().unwrap();
+
+    println!("Result: {}", hex::encode(invoke_result));
+    println!("Gas Used: {}", invoke_res.msg_receipt.gas_used);
+
+    Ok(())
```

Comment:
I'd generally _return_ these as results then print at the top, but eh, good enough to start with.